### PR TITLE
[Merged by Bors] - Relicense bors checks

### PR DESCRIPTION
**Pull request by DJMcNab**
# Objective

Currently we can sometimes allow PRs by people who haven't agreed to the relicense to get merged into main.

E.g. https://github.com/bevyengine/bevy/pull/2445

## Solution

This adds a check to ensure that this doesn't happen, by ensuring that bors doesn't relicense said PRs.

As a bonus, it also adds config to automatically label new PRs as needing checking, to ensure that we do the verification until we merge the new license.

### CODE REVIEW

**Review by mockersf — APPROVED** [review]

**Comment by mockersf** [comment]
Not a fan of the auto label but it makes sense and shouldn't be for long

**Comment by DJMcNab** [comment]
Ok, here's a weird thing - this PR hasn't been automatically labelled with `S-Needs-Triage`. That seems *strange*

**Comment by DJMcNab** [comment]
Yes, indeed. The automatic labelling is only temporary. I'd rather make sure that we don't have any false approvals again, so it's best to check.

**Comment by mockersf** [comment]
> Ok, here's a weird thing - this PR hasn't been automatically labelled with `S-Needs-Triage`. That seems _strange_

That shouldn't come from the change in this PR as the label workflow is `pull_request_target` so it's using config file from main... still strange

**Comment by DJMcNab** [comment]
Ok, this is 'expected' - actions/labeler doesn't match files/folders starting with a `.` from a glob pattern (because it doesn't have https://github.com/isaacs/minimatch#dot set)

**Review by NiklasEi — APPROVED** [review]
Definitely better to remove a few premature labels than to merge a PR and then have to run after the contributor to approve the license change :+1:

**Pull request closed without merging**
